File: app/src/main/java/com/benhero/glstudio/base/GLObjectUtils.java
package com.benhero.glstudio.base;

/**
 * OpenGL绘制对象工具类
 *
 * @author Benhero
 */
public class GLObjectUtils {

    public static void setToCenter(GLObject object) {
        setToCenterX(object);
        setToCenterY(object);
    }

    public static void setToCenterX(GLObject object) {
        object.setX((object.getParentWidth() - object.getWidth()) / 2);
    }

    public static void setToCenterY(GLObject object) {
        object.setY((object.getParentHeight() - object.getHeight()) / 2);
    }
}

File: app/src/main/java/com/benhero/glstudio/base/GLObjectUtils.kt
package com.benhero.glstudio.base

/**
 * OpenGL绘制对象工具类
 *
 * @author Benhero
 */
object GLObjectUtils {

    fun setToCenter(`object`: GLObject) {
        setToCenterX(`object`)
        setToCenterY(`object`)
    }

    fun setToCenterX(`object`: GLObject) {
        `object`.x = (`object`.parentWidth - `object`.width) / 2
    }

    fun setToCenterY(`object`: GLObject) {
        `object`.y = (`object`.parentHeight - `object`.height) / 2
    }
}
